src/odemis/driver/focustracker.py

```diff
@@ -39,164 +43,162 @@ def __init__(self, name, role, channel, node_idx, **kwargs):
-            self.tracking = model.BooleanVA(self._tracking_sdo.raw, getter=self.get_tracking, setter=self.set_tracking)
+        self._metadata[model.MD_GAIN_P] = self._proportional_gain_sdo.raw
+        self._metadata[model.MD_GAIN_I] = self._integral_gain_sdo.raw
+        self._metadata[model.MD_GAIN_I] = self._derivative_gain_sdo.raw
+        self.targetPosition = model.FloatContinuous(self._target_pos_sdo.raw, TARGET_POSITION_RANGE, unit="m",
+                                                    getter=self._get_target_pos, setter=self._set_target_pos)
+        self.position = model.FloatVA(self._position_sdo.raw, unit="m", readonly=True, getter=self._get_position)
+        self.tracking = model.BooleanVA(self._tracking_sdo.raw, getter=self._get_tracking,
+                                        setter=self._set_tracking)
 
     def terminate(self):
         """Disconnect from CAN bus."""
         if self.network:
+            self._set_tracking(False)
             self.network.sync.stop()
             self.network.disconnect()
             self.network = None
 
     def updateMetadata(self, md):
         if model.MD_GAIN_P in md:
-            md[model.MD_GAIN_P] = self.set_proportional(md[model.MD_GAIN_P])
+            md[model.MD_GAIN_P] = self._set_proportional(md[model.MD_GAIN_P])
         if model.MD_GAIN_I in md:
-            md[model.MD_GAIN_I] = self.set_integral(md[model.MD_GAIN_I])
+            md[model.MD_GAIN_I] = self._set_integral(md[model.MD_GAIN_I])
         if model.MD_GAIN_D in md:
-            md[model.MD_GAIN_D] = self.set_derivative(md[model.MD_GAIN_D])
+            md[model.MD_GAIN_D] = self._set_derivative(md[model.MD_GAIN_D])
         super(FocusTrackerCO, self).updateMetadata(md)
 
-    def get_available_nodes(self):
-        """
-        return (list): a list of id's of nodes in the network
-        """
-        return self.network.scanner.nodes
-
-    def get_object_dictionary(self):
+    def _get_position(self):
         """
-        return (dict): the object dictionary of the focus tracker.
+        return (float): The current position of the laser on the linear ccd.
         """
-        return self.node.object_dictionary
+        return self._position_sdo.raw * 1e-6
 
-    def get_current_pos(self):
+    def _get_target_pos(self):
         """
-        return (float): The current position of the laser on the linear ccd.
+        return (float): The target position in meters of the laser on the linear ccd.
         """
-        return self._current_pos_sdo.raw * 1e-6
+        return self._target_pos_sdo.raw * 1e-6
 
-    def get_target_pos(self):
+    def _set_target_pos(self, value):
         """
+        value (float, int): The target position in meters.
         return (float): The target position of the laser on the linear ccd.
         """
-        return self._target_pos_sdo.raw * 1e-6
+        self._target_pos_sdo.raw = value * 1e6
+        return value
```

Comment:
On the hardware it is possible to set it to negative values and whatever range. An `IndexError` is raised when you try to set is with a value out of range, because that's what the `FloatContinuous` VA does